libazureinit/src/provision/ssh.rs

```diff
@@ -43,123 +45,225 @@ pub(crate) fn provision_ssh(
     Ok(())
 }
 
-pub(crate) fn get_authorized_keys_path_from_sshd() -> io::Result<Option<PathBuf>>
-{
-    let sshd_output = Command::new("sshd").arg("-T").output()?;
-    if !sshd_output.status.success() {
-        return Err(io::Error::new(
-            io::ErrorKind::Other,
-            format!("sshd -T failed with status: {}", sshd_output.status),
-        ));
+fn get_sshd_g_output() -> io::Result<Output> {
+    Command::new("sshd").arg("-G").output()
+}
+
+fn get_authorized_keys_path_from_sshd(
+    sshd_config_command_runner: impl Fn() -> io::Result<Output>,
+) -> Option<String> {
+    let output: Output = match run_sshd_command(sshd_config_command_runner) {
+        Some(output) => output,
+        None => {
+            info!("Unable to determine authorizedkeysfile from sshd -G");
+            return None;
+        }
+    };
+
+    match extract_authorized_keys_file_path(&output.stdout) {
+        Some(path) => Some(path),
+        None => {
+            info!("No authorizedkeysfile in sshd configuration");
+            None
+        }
     }
```

Comment:
You could also write this:
```Rust
let path = extract_authorized_keys_file_path(&output.stdout);
if path.is_none() {
    info!("No authorizedkeysfile setting found in sshd configuration");
}
path
```

it's equivalent and not a big deal, it's just unusual to have all the arms of a match just return the same type thing they match and it makes it clear you're returning what you get from the function, just logging if it's none.

Reply:
I like it! Also bumped this to error level as it would be unexpected.